src/v/cluster/config_frontend.cc

```diff
@@ -87,10 +88,36 @@ ss::future<config_frontend::patch_result> config_frontend::do_patch(
           .errc = errc::waiting_for_recovery, .version = config_version_unset};
     }
 
-    auto data = cluster_config_delta_cmd_data();
+    // do not transform values if the cluster is in the middle of an upgrade.
+    // our aliases might be primary names for the other nodes
+    auto in_mixed_state = [&] {
+        auto& ft = _features.local();
+        return ft.get_active_version() != ft.get_latest_logical_version();
+    }();
 
-    data.upsert = std::move(update.upsert);
-    data.remove = std::move(update.remove);
+    // try to convert aliases to primary names, or no op
+    auto normalize_prop_name = [&](std::string_view prop_name) {
+        if (in_mixed_state) {
+            return ss::sstring{prop_name};
+        }
+        if (config::shard_local_cfg().contains(prop_name)) {
+            return ss::sstring{config::shard_local_cfg().get(prop_name).name()};
+        }
+        return ss::sstring{prop_name};
+    };
+
+    auto normalized_upsert
+      = update.upsert | std::views::transform([&](auto in) {
+            return cluster_property_kv{normalize_prop_name(in.key), in.value};
+        })
+        | std::views::common;
+    auto normalized_remove = update.remove
+                             | std::views::transform(normalize_prop_name)
+                             | std::views::common;
+
```

Comment:
nit: it feels like we're not really saving LOC or reducing complexity vs what seems like it could've been a couple for loops. I don't feel too strongly about it, but without exposure to this library it took longer to understand what this was doing, vs a simple loop

Reply:
in c++23 this becomes def shorter, could be

```cpp
 .upsert= {std::from_range, update.upsert | std::views::transform([&](auto in) {
            return cluster_property_kv{normalize_prop_name(in.key), in.value};
        })}
```
the annoying bit is ::common to adapt the range the current constructor of vector. 

will have a look again at this, I don't feel too strongly about this but I'm on the opposite side :D